portal/static/portal/js/passwordStrength_new.js

```diff
@@ -0,0 +1,97 @@
+
+You are not authorised to use the name Ocado (or any of its trade names) or
+the names of any author or contributor in advertising or for publicity purposes
+pertaining to the distribution of this program, without the prior written
+authorisation of Ocado.
+
+Any propagation, distribution or conveyance of this program must include this
+copyright notice and these terms. You must not misrepresent the origins of this
+program; modified versions of the program must be marked as such and not
+identified as the original program.
+*/
+var PASSWORD_TYPE = '';
+var PASSWORD_FIELD_ID = '';
+var password_field = '';
+
+$(function() {
+    var updateFunction = (PASSWORD_TYPE == 'INDEPENDENT_STUDENT') ? updateIndependentStudentPasswordStrength : updateTeacherPasswordStrength;
+
+    password_field = $('#' + PASSWORD_FIELD_ID);
+    password_field.on('keydown', updateFunction);
+    password_field.on('paste', updateFunction);
+    password_field.on('cut', updateFunction);
+    updateFunction();
+});
+
+var teacher_password_strengths = [
+    { name: 'No password!', colour: '#FF0000' },
+    { name: 'Password too weak', colour: '#DBA901' },
+    { name: 'Strong password', colour: '#088A08' }
+];
+
+function updateTeacherPasswordStrength() {
+    // The reason for the timeout is that if we just got $('#...').val() we'd get the
+    // old value before the keypress / change. Apparently even jQuery itself implements
+    // things this way, so maybe there is no better workaround.
+
+    setTimeout(function() {
+        var password = $('#' + PASSWORD_FIELD_ID).val();
+
+        var strength = 0;
+        if (password.length > 0) { strength++; }
+        if (password.length > 8 && !(password.search(/[A-Z]/) === -1 || password.search(/[a-z]/) === -1 || password.search(/[0-9]/) === -1)) { strength++; }
+
+        $('.password-strength-sign').css('background-color', teacher_password_strengths[strength].colour);
+        $('.password-strength-text').html(teacher_password_strengths[strength].name);
+    });
+}
+
+var independent_student_password_strengths = [
+    { name: 'Password quality', colour: '' },
+    { name: 'Not long enough', colour: '#DBA901' },
+    { name: 'Good password', colour: '#088A08' }
+];
+
+function updateIndependentStudentPasswordStrength() {
+    setTimeout(function() {
+        var password = $('#' + PASSWORD_FIELD_ID).val();
+
+        var strength = 0;
+        if (password == '') { strength = 0; }
+        else if (password.length < 6) { strength = 1; }
+        else if (password.length >= 6) { strength = 2; }
+
+        if (password == '') { $('.password-strength-bar-container').hide(); $('.password-strength-text').hide(); }
+        else { $('.password-strength-bar-container').show(); $('.password-strength-text').show(); }
+
+        $('.password-strength-bar').css('width', strength / 2 * 100 + '%');
+        $('.password-strength-bar').css('background-color', independent_student_password_strengths[strength].colour);
```

Comment:
I see that you've copied this way of doing things from the existing code, but a cleaner way of doing it would be to use a class, and define the background-color in CSS or Sass vars.  jQuery's toggleClass or addClass/removeClass would replace this declaration.